### ISSUE TITLE
`SelectionArea` breaks for TalkBack 

### ISSUE THREAD

**Issue by nilsreichardt** [flutter/flutter#104557]
## Description 
Wrapping a `SelectionArea` widget around multiple Text widgets breaks the selection for TalkBalk. TalkBalk is still able to read the text but I'm unable to select a `Text` widget inside the tree.

### With `SelectionArea`
https://user-images.githubusercontent.com/24459435/170145265-6f4b3b4b-2911-4912-818e-056f2238f598.MOV

### Without `SelectionArea` (how it should be)
https://user-images.githubusercontent.com/24459435/170145239-a84091bb-135d-466a-bf7f-359861b07287.MOV

I used a Google Pixel 4a with Android.

<details>
<summary>My app</summary>

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// This example shows how to make a screen selectable..

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      // You need to add the "SelectionArea" to make the text selectable.
      home: SelectionArea(
        child: Scaffold(
          body: Padding(
            padding: const EdgeInsets.all(16),
            child: Center(
              child: Column(
                mainAxisSize: MainAxisSize.min,
                children: <Widget>[
                  Text(
                    'Fast',
                    style: Theme.of(context).textTheme.titleLarge,
                  ),
                  const Text(
                    'Flutter code compiles to ARM or Intel machine code as well as JavaScript, for fast performance on any device.',
                  ),
                  ElevatedButton(
                    onPressed: () {},
                    child: const Text('Try it in DartPad'),
                  )
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

cc: @chunhtai I think this should get a higher priority because TalkBalk is an import accessibility feature.

**Comment by chunhtai** [comment]
It looks like the text gets merge into the selectableRegion's gesture detector. It seems like we can add Semantics widget with explicitChildNodes= true, but that will prevent them from merging into parent....
